src/coreclr/src/vm/peimage.inl

```diff
@@ -498,7 +498,8 @@ inline PTR_PEImage PEImage::FindByPath(LPCWSTR pPath)
 #else
     DWORD dwHash = CaseHashHelper(pPath, (COUNT_T) wcslen(pPath));
 #endif
-   return (PEImage *) s_Images->LookupValue(dwHash, &locator);
+    dwHash = (dwHash << 1) | lookInBundle;
```

Comment:
You do not really need to include `lookInBundle` in the hashcode. It is unlikely that there is going to be collision on the same path is pretty unlikely.

However, you do need to make sure to compare `lookInBundle` in `CompareImage` that is `s_Images` equals function. Otherwise, you can end up with intermittent wrong results due to hash collisions.